src/main/java/teammates/ui/pagedata/FeedbackSubmissionEditPageData.java

```diff
@@ -268,4 +268,20 @@ private FeedbackSubmissionEditQuestion createQuestion(FeedbackQuestionAttributes
 
         return responses;
     }
+
+    /**
+     * Returns true if there is an existing response in the form.
+     */
+    public boolean getIsResponsePresent() {
```

Comment:
More on code quality discussion:
`StudentFeedbackSubmissionEditQuestionsWithResponses`/`FeedbackSubmissionEditResponse ` are something related to UI while the method actually should check whether the data exist or not. If u take a look at `FeedbackSubmissionEditPageData`. You will find that we add fake `FeedbackSubmissionEditResponse` and set `isExisitingResponse` to `false` to just make the UI correctly displayed. My suggestion is to check the `bundle` instead of the `questionsWithResponses` as `questionsWithResponses` is generated by `bundle` (Single point of truth). What do u think?

Reply:
Hi @xpdavid 

Thanks for reviewing :) You have a good point! 

This is how I will do it if we are checking from bundle instead.

```
    public boolean getIsResponsePresent() {
        for (FeedbackQuestionAttributes questionAttributes : bundle.getSortedQuestions()) {
            List<FeedbackResponseAttributes> existingResponses =
                    bundle.questionResponseBundle.get(questionAttributes);
            for (FeedbackResponseAttributes existingResponse : existingResponses) {
                if (isResponseRecipientValid(existingResponse)) {
                    return true;
                }
            }
        }
        return false;
    }
```
How does this look?

Reply:
- I think `existingResponses` doesn't fit the context here. Could just use `responses`

- `getSortedQuestions()` -> `bundle.questionResponseBundle`/`bundle.getQuestionResponseBundle`: `getSortedQuestions()` also seems weired as there is nothing about order here. You can just check the do a foreach loop to the map and check whether the response list is empty or not.

Reply:
Hi @xpdavid 

I would like to clarify if you are suggesting an approach that uses lambda expression as you mentioned about `forEach`. Nevertheless, I tried figuring out the solution using `bundle.questionResponseBundle.forEach()` but it isn't very direct as `forEach` itself returns void.

This is my new attempt using an enhanced for loop :smile:  

```
    public boolean getIsResponsePresent() {
        for (Map.Entry<FeedbackQuestionAttributes, List<FeedbackResponseAttributes>>
                entry : bundle.questionResponseBundle.entrySet()) {
            if (!entry.getValue().isEmpty()) {
                return true;
            }
        }
        return false;
    }
```